src/passes/StripTargetFeatures.cpp

```diff
@@ -19,11 +19,14 @@
 namespace wasm {
 
 struct StripTargetFeatures : public Pass {
+  bool isStripped = false;
+  StripTargetFeatures(bool isStripped) : isStripped(isStripped) {}
   void run(PassRunner* runner, Module* module) override {
-    module->hasFeaturesSection = false;
+    module->hasFeaturesSection = !isStripped;
   }
 };
 
-Pass* createStripTargetFeaturesPass() { return new StripTargetFeatures(); }
+Pass* createStripTargetFeaturesPass() { return new StripTargetFeatures(true); }
+Pass* createEmitTargetFeaturesPass() { return new StripTargetFeatures(false); }
```

Comment:
Do we actually emit such a section if one didn't exist before?

A test would prove it works ;)